src/embed.ts

```diff
@@ -83,21 +72,5 @@ export async function embed(
 
   const embeddings = await all(res.value);
 
-  const e = embeddings[0];
-  if (!e) {
-    return result.Err({
-      message: 'Got response with no embedding',
-      statusCode: 200,
-    });
-  }
-
-  if (embeddings.length > 1) {
-    return result.Err({
-      message:
-        'Got response with multiple embeddings, expected 1, please contact support',
-      statusCode: 200,
-    });
-  }
-
-  return result.Ok({ embedding: e });
+  return result.Ok({ embeddings: embeddings.flat() });
```

Comment:
Flat shouldn't be needed here, I think I wrote the code here poorly it doesn't make it clear why embeddings was an array. It's an array of responses, it's handling json steaming, but we expect a single json object.

We just need the 0th element here. I can take a stab at making it clearer from makeRequest side